pymzml/spec.py

```diff
@@ -1058,6 +1058,37 @@ def peaks(self, peak_type):
             peaks.sort(key=itemgetter(0))
         return peaks
 
+    @lru_cache()
+    def get_array(self, arr_name):
+        array_params = self._get_encoding_parameters(arr_name)
+        array = self._decode(*array_params)
+        if len(array) == 0:
+            array = None
+            _ = self.get_all_arrays_in_spec(not_found_array=arr_name)
+        return array
+
+    def get_tims_tof_ion_mobility(self, array_name='mean inverse reduced ion mobility array'):
+        arr = self.get_array(array_name)
+        if arr is None:
+            _ = self.get_all_arrays_in_spec(not_found_array=array_name)
+        return arr
```

Comment:
Hi @MKoesters !

Is this also working for Waters ion mobility files?
Or should we use a more general function name e.g. `get_ion_mobility`?

Reply:
I tried to convert some of the Water Synapt G2 files, you mentioned in the issue.
However I did not manage to convert them to mzML such that they contain a ion mobility array.
This mean, I don't know what the name of this array for waters instruments is.
However, if you know the name, you can get it by `spec.get_array(array_name)`
For the tims tof pro, the array name is `mean inverse reduced ion mobility array`, but according to psi-ms obo, there are multiple possible names for ion mobility arrays.

Reply:
I see...
Water Synapt G2 can aquire ion mobility + DDA and ionmobilty + DIA, which probably also have different names in mzML...
I will see what I can do when I got some new files to test.
If I got some solution I will try to create a new pull request.

Reply:
To get Waters files to produce an ion mobility array with ProteoWizard, you need to enable the "combine ion mobility spectra" option. If you do that, you'll get a "raw ion mobility array" over a spectrum instead of a bunch of spectra with associated with discrete "drift time" values. There are some caveats about what the points in a combined spectrum mean though.

There are also these other CV terms for ion mobility arrays:
- raw inverse reduced ion mobility array
- mean drift time array
- mean ion mobility array
- ion mobility array

You could cycle through them until one hits and return that one. It's unlikely you'll get multiple flavors of ion mobility instrumentation in the same measurement, though ion mobility arrays are synthetic so who knows. You still have to do a lot of cleaning since the ion mobility coordinates aren't centroids.

Reply:
@mobiusklein 
Can you tell me which exact version of ProteoWizard you are using? 
Many thanks!

Reply:
I'm using MSConvert 3.0.20236.400e34473, but this feature has been around for at least most of this year. The tricky bit is it's not a filter, it's a dedicated CLI option `--combineIonMobilitySpectra` or a checkbox in the part of MSConvert's GUI that you're liable to never look:
![image](https://user-images.githubusercontent.com/1426041/97065057-bf9e1800-1578-11eb-8c0d-35cbe00178ad.png)

The combined spectra you'll get from a Waters file will be profile mode in both the m/z and drift time since they [don't support peak picking in IMS spectra](https://github.com/ProteoWizard/pwiz/blob/master/pwiz/data/vendor_readers/Waters/SpectrumList_Waters.cpp#L198-L202). I think they'll be sorted by drift time then m/z, so you'll need to keep that in mind when accessing or manipulating them.

Reply:
Many thanks!